fix(react): rework the react client to support re-render (#451)
flipt-io/flipt-client-sdks: flipt-client-react/README.md

}
```

#### useFliptBoolean Hook

> The `useFliptBoolean` hook is a convenience hook that returns the boolean evaluation that it can be used in a functional component that is wrapped in a `FliptProvider`. `useFliptBoolean` requires the fallback value if it's impossible to evaluate the flag at the moment.

Reviewer: ```suggestion
The `useFliptBoolean` hook simplifies the process of evaluating a feature flag as a boolean. This hook must be used within a functional component that is wrapped by the `FliptProvider` context.
A default value is returned if the feature flag cannot be evaluated at the current time (e.g., due to network issues or missing data).
```

```tsx
import { useFliptBoolean } from '@flipt-io/flipt-client-react';

function MyComponent() {
  const result = useFliptBoolean('my-flag', false, 'user-123', {
    /*additional context*/
  });

>   // Use the client to evaluate flags

Reviewer: ```suggestion
```

  const handleCheckFlag = async () => {
    console.log('Flag evaluation result:', result);
  };

  return (
    <div>
      <button onClick={handleCheckFlag}>Check Flag</button>
    </div>
  );
}
```

#### useFliptVariant Hook

> The `useFliptVariant` hook is a convenience hook that returns the variant evaluation that it can be used in a functional component that is wrapped in a `FliptProvider`. `useFliptVariant` requires the fallback value if it's impossible to evaluate the flag at the moment.

Reviewer: ```suggestion
The `useFliptVariant` hook simplifies the process of evaluating a variant feature flag. This hook must be used within a functional component that is wrapped by the `FliptProvider` context.
A default value is returned if the feature flag cannot be evaluated at the current time (e.g., due to network issues or missing data).
```

```tsx
import { useFliptVariant } from '@flipt-io/flipt-client-react';

function MyComponent() {
  const result = useFliptVariant('my-flag', 'fallback', 'user-123', {
    /*additional context*/
  });

>   // Use the client to evaluate flags

Reviewer: ```suggestion
```

}
```

### useFliptSelector Hook

> The `useFliptSelector` hook allows the direct access to the Flipt client so that it can be used in a functional component that is wrapped in a `FliptProvider`.

Reviewer: ```suggestion
The `useFliptSelector` hook allows direct access to the Flipt client so that it can be used in a functional component that is wrapped in a `FliptProvider`.
```

> This is useful for more complex evaluations or in cases where the Flipt client other methods should be called.

Reviewer: ```suggestion
This is useful for more complex evaluations or in cases where you wish to call other methods on the Flipt client.
```

> [!CAUTION]

> > `flipt-client-react` heavily depends on the `useSyncExternalStore` and it has the same [caveats](https://react.dev/reference/react/useSyncExternalStore#caveats)

Reviewer: ```suggestion
> `flipt-client-react` heavily depends on the `useSyncExternalStore` which has some [caveats](https://react.dev/reference/react/useSyncExternalStore#caveats)
```

- `options`: An instance of the `ClientOptions` type that supports several options for the client. The structure is:
  - `url`: The URL of the upstream Flipt instance. If not provided, the client will default to `http://localhost:8080`.
  - `authentication`: The authentication strategy to use when communicating with the upstream Flipt instance. If not provided, the client will default to no authentication. See the [Authentication](#authentication) section for more information.

>   - `updateInterval`: The polling interval for fetching new state from Flipt. Default set `120` seconds. To disable the polling please use `0`.

Reviewer: ```suggestion
  - `updateInterval`: The polling interval for fetching new state from Flipt. Set to `120` seconds by default. A `0` value disables polling completely after the initial fetch.
```
